Python/import.c

```diff
@@ -4367,7 +4371,9 @@ _PyImport_LazyImportModuleLevelObject(PyThreadState *tstate,
             Py_DECREF(filter);
             Py_DECREF(abs_name);
             return NULL;
-        } else if (modname == NULL) {
+        }
+        if (modname == NULL) {
+            assert(!PyErr_Occurred());
```

Comment:
Actually this one is not needed because this means that `PyDict_GetItemRef` returned 0 or 1, and thus `modname` is already set. However for readability purposes, I think it's better to separate the ifs.